MarsRover.Tests/MapShould.cs

```diff
@@ -19,5 +17,27 @@ public void GetSquareAtLocation_ReturnCorrectSquare_WhenGivenALocation(List<Squa
 
             result.Should().BeEquivalentTo(expected);
         }
+
+        [Theory]
+        [MemberData(nameof(MapShouldTestData.HasObstacleTrueTestData), MemberType = typeof(MapShouldTestData))]
+        public void HasObstacle_ReturnsTrue_GivenLocation(int x, int y)
+        {
+            Map map = MapShouldTestData.SetMap();
+
+            var result = map.HasObstacle(new Location(x, y));
+
+            result.Should().BeTrue();
+        }
+
+        [Theory]
+        [MemberData(nameof(MapShouldTestData.HasObstacleFalseTestData), MemberType = typeof(MapShouldTestData))]
+        public void HasObstacle_ReturnsFalse_GivenLocation(int x, int y)

```

Comment:
```suggestion
        public void HasObstacle_ReturnsFalse_GivenLocationWithNoObstacle(int x, int y)

```

Reply:
differentiate what the 2 tests are testing, otherwise the test names appear contradictory